packages/framework/client-logger/app-insights-logger/src/fluidAppInsightsLogger.ts

```diff
@@ -104,9 +127,47 @@ export class FluidAppInsightsLogger implements ITelemetryBaseLogger {
 		return shouldSendEvent;
 	}
 
+	/**
+	 * Checks if a given {@link ITelemetryBaseEvent} conforms to any of the provided {@link TelemetryFilter} rules.
+	 *
+	 * - If a {@link TelemetryFilter} specifies both a `category` and a `namespace`, the event must match both.
+	 * - If only a `category` or `namespace` is provided, the event should match either one of them.
+	 * - If a `namespace` pattern exception is specified in the {@link TelemetryFilter}, the event should not match the exception pattern.
+	 *
+	 * @param event - The telemetry event to check against the filters.
+	 *
+	 * @returns boolean `true` if the event matches any filter, otherwise `false`.
+	 */
 	private doesEventMatchFilter(event: ITelemetryBaseEvent): boolean {
 		for (const filter of this.config.filtering.filters ?? []) {
-			if (filter.category === event.category) {
+			let matches = true;
+
+			// If the filter has a category and it doesn't match the event's category, skip to next filter
+			if (filter.category && event.category !== filter.category) {
```

Comment:
Note that this will treat an empty `filter.category` string value the same as if were `undefined`. I suspect this probably isn't what we want. I would expect that if a user specified an empty string as their category, it would match nothing. In which case, this should be:
```suggestion
			if (filter.category !== undefined && event.category !== filter.category) {
```

If that's not the case, and we do want to treat empty string as "no category provided", we should be sure to document that explicitly in the property docs.

Reply:
You are right. After moving to a list of categories this shouldn't be an issue any more.